docker-compose.yml

```diff
@@ -39,6 +39,7 @@ services:
     ports:
       - "27017:27017"
       - "28017:28017"
+    command: ["--replSet", "mongo-replica-set"]
```

Comment:
This command seems to be causing problems with Mongoid on applications.

On Slack people have reported that you can't run `make publisher` and we seem to have traced it to this change.

E.g:

```
$ git checkout m1-compatibility~1; govuk-docker down; govuk-docker run publisher-lite bundle exec rails db:setup
HEAD is now at b5b94d7 Use elasticsearch-7 for for M1 compatibility
docker-compose -f [...] down
[+] Running 4/3
 ⠿ Container govuk-docker_publisher-lite_run_fbbd6effc69e  Removed                                                                                                                                                                                                                                                                                                     0.0s
 ⠿ Container govuk-docker-mongo-3.6-1                      Removed                                                                                                                                                                                                                                                                                                     0.3s
 ⠿ Container govuk-docker-redis-1                          Removed                                                                                                                                                                                                                                                                                                     0.2s
 ⠿ Network govuk-docker_default                            Removed                                                                                                                                                                                                                                                                                                     0.1s
docker-compose -f [...] run publisher-lite bundle exec rails db:setup
[+] Running 1/1
[+] Running 3/3k-docker_default  Created                                                                                                                                                                                                                                                                                                                               0.0s
 ⠿ Network govuk-docker_default        Created                                                                                                                                                                                                                                                                                                                         0.0s
 ⠿ Container govuk-docker-redis-1      Created                                                                                                                                                                                                                                                                                                                         0.1s
 ⠿ Container govuk-docker-mongo-3.6-1  Created                                                                                                                                                                                                                                                                                                                         0.1s
[+] Running 2/2
 ⠿ Container govuk-docker-redis-1      Started                                                                                                                                                                                                                                                                                                                         0.3s
 ⠿ Container govuk-docker-mongo-3.6-1  Started                                                                                                                                                                                                                                                                                                                         0.4s
** Invoke db:setup (first_time)
** Invoke db:create (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute db:create
** Invoke db:mongoid:create_indexes (first_time)
** Invoke environment
** Invoke db:mongoid:load_models (first_time)
** Execute db:mongoid:load_models
** Execute db:mongoid:create_indexes
** Invoke db:seed (first_time)
** Invoke environment
** Execute db:seed
** Execute db:setup
```

vs:

```
$ git checkout m1-compatibility; govuk-docker down; govuk-docker run publisher-lite bundle exec rails db:setup
Previous HEAD position was b5b94d7 Use elasticsearch-7 for for M1 compatibility
Switched to branch 'm1-compatibility'
Your branch is up to date with 'origin/m1-compatibility'.
docker-compose -f [...] down
[+] Running 4/3
 ⠿ Container govuk-docker_publisher-lite_run_c698634de2a2  Removed                                                                                                                                                                                                                                                                                                     0.0s
 ⠿ Container govuk-docker-mongo-3.6-1                      Removed                                                                                                                                                                                                                                                                                                     0.3s
 ⠿ Container govuk-docker-redis-1                          Removed                                                                                                                                                                                                                                                                                                     0.2s
 ⠿ Network govuk-docker_default                            Removed                                                                                                                                                                                                                                                                                                     0.1s
docker-compose -f [...] run publisher-lite bundle exec rails db:setup
[+] Running 1/1
[+] Running 3/3k-docker_default  Created                                                                                                                                                                                                                                                                                                                               0.0s
 ⠿ Network govuk-docker_default        Created                                                                                                                                                                                                                                                                                                                         0.0s
 ⠿ Container govuk-docker-redis-1      Created                                                                                                                                                                                                                                                                                                                         0.1s
 ⠿ Container govuk-docker-mongo-3.6-1  Created                                                                                                                                                                                                                                                                                                                         0.1s
[+] Running 2/2
 ⠿ Container govuk-docker-mongo-3.6-1  Started                                                                                                                                                                                                                                                                                                                         0.4s
 ⠿ Container govuk-docker-redis-1      Started                                                                                                                                                                                                                                                                                                                         0.3s
** Invoke db:setup (first_time)
** Invoke db:create (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute db:create
** Invoke db:mongoid:create_indexes (first_time)
** Invoke environment
** Invoke db:mongoid:load_models (first_time)
** Execute db:mongoid:load_models
** Execute db:mongoid:create_indexes
rails aborted!
Mongo::Error::NoServerAvailable: No primary server is available in cluster: #<Cluster topology=Unknown[mongo-3.6:27017] servers=[#<Server address=mongo-3.6:27017 GHOST>]> with timeout=30, LT=0.015

Tasks: TOP => db:setup => db:mongoid:create_indexes
(See full trace by running task with --trace)
```

Reply:
Oddly `make publisher` has been working fine for me on this branch, but that is after tearing down my entire Docker environment in debug tools.

Reply:
I get this issue too now: 

```
Mongo::Error::NoServerAvailable: No primary server is available in cluster: #<Cluster topology=Unknown[mongo-3.6:27017] servers=[#<Server address=mongo-3.6:27017 GHOST>]> with timeout=30, LT=0.015
```

Reply:
@KludgeKML since this is causing problems. I've pushed this branch with b7110cb removed from it.

I've pushed up a new branch `m1-compatibility-keith` that still has it on it.

Reply:
Why do we need it to replicate locally?

Reply:
It's necessary for router development (but not for router work). Resolution is we merge this, and the handful of people doing active router development live with a small workaround until we can get router updated (coming soon).